editor/osm_feature_matcher.cpp

```diff
@@ -49,54 +68,78 @@ vector<m2::PointD> GetWaysGeometry(pugi::xml_document const & osmResponse,
   return result;
 }
 
+vector<m2::PointD> GetRelationsGeometry(pugi::xml_document const & osmResponse,
+                                        pugi::xml_node const & relation)
+{
+  vector<m2::PointD> result;
+  ForEachRelationsNode(osmResponse, relation, [&result](XMLFeature const & xmlFt)
```

Comment:
Но алгоритм не сработает, если вей не догрузится полностью. Верно?

> On Mar 21, 2016, at 15:38, mgsergio <notifications@github.com> wrote:
> 
> In editor/osm_feature_matcher.cpp <https://github.com/mapsme/omim/pull/2372#discussion_r56814982>:
> 
> > @@ -49,54 +68,78 @@ vector<m2::PointD> GetWaysGeometry(pugi::xml_document const & osmResponse,
> >    return result;
> >  }
> >  
> > +vector<m2::PointD> GetRelationsGeometry(pugi::xml_document const & osmResponse,
> > +                                        pugi::xml_node const & relation)
> > +{
> > +  vector<m2::PointD> result;
> > +  ForEachRelationsNode(osmResponse, relation, [&result](XMLFeature const & xmlFt)
> Веи и релейшены проверяются в общей куче. Победит тот, уоторый наберет больше очков.
> Случай, когда один вей (тот, что должен заматчиться) является частью релейшена,
> обработается корректно: релейшн получит меньше очков, так как у него будет много лишних точек. Алгоритм вернет желаемый вей.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2372/files/904ccf66b255cd28dd6543a812461901ba00c06c#r56814982>